codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -341,4 +395,12 @@ public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
         super.handlerRemoved(ctx);
         cleanup();
     }
+
+    public final int blockSize() {
+        return blockSize;
+    }
+
+    public final int currentBlockLength() {
+        return currentBlockLength;
```

Comment:
can you explain what this is? Can we just use `buffer.readableBytes()` instead?

Also some places where it is used seem suspect. For example line 257...

```java
out.setBytes(idx + HEADER_LENGTH, buffer, 0, currentBlockLength);
```
Should the following changes be made here?
`buffer` -> `buf`
`0` -> `buf.readerIndex()`
`currentBlockLength` -> `buf.readableBytes()`